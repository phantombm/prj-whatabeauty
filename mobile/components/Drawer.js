import React, { Component } from 'react';
import Drawer from 'react-native-drawer'

import DrawerContent from '../components/DrawerContent';

export default class _Drawer extends Component {
  openDrawer = () => {
    this.drawer.open();
  };

  closeDrawer = () => {
    this.drawer.close();
  };

  render() {
    return (
      <Drawer
        acceptTap
        type="overlay"
        tweenDuration={ 100 }
        openDrawerOffset={ 0.3 }
        ref={ (ref) => { this.drawer = ref; } }
        content={ <DrawerContent closeDrawer={ this.closeDrawer } /> }
        styles={{ drawer: { backgroundColor: '#ffffff' }, mainOverlay: { backgroundColor: '#000000', opacity: 0 } }}
        tweenHandler={ (ratio) => ({ main: { opacity: ( 2 - ratio ) / 2 }, mainOverlay: { opacity: ratio * 0.8 } }) }
      >{ this.props.children }</Drawer>
    );
  }
}
