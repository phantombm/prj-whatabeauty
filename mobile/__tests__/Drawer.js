import React from 'react';
import 'react-native';
import testRenderer from 'react-test-renderer';

import Drawer from '../components/Drawer';

test('renders correctly', () => {
  const tree = testRenderer.create(
    <Drawer />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
