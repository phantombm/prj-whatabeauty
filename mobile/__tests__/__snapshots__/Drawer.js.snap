// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "borderWidth": 0,
          "left": 0,
          "position": "absolute",
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "#000000",
            "opacity": 0,
          },
        ]
      }
    />
  </View>
  <View
    elevation={0}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderWidth": 0,
          "left": -525,
          "position": "absolute",
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 525,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
          >
            Main
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
          >
            Links
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
